front/app/components/ProjectAndFolderCards/ProjectAndFolderCards.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useEffect, useMemo } from 'react';
```

Comment:
Yeah I know, but I didn't really know what else to call this one.

Also, they can't be the same file, since `useAdminPublications` in this component depends on the output of `useAdminPublicationsStatusCount` in the `index` file. Since the output `useAdminPublicationsStatusCount` is only available async, we don't know yet what `publicationStatusFilter` we need to give `useAdminPublications` when we call it. If you really want to put them in the same component you would have to give `useAdminPublications` some random `publicationStatusFilter` and then have it refetch data when the actual `publicationStatusFilter` is available. This is super messy and will create unnecessary requests

Reply:
So you will have in one component:

```
const { counts, onChangeAreas } = useAdminPublicationsStatusCounts({
    publicationStatusFilters: allowedPublicationStatuses,
    rootLevelOnly: true,
    removeNotAllowedParents: true,
});

const [
    publicationStatusesForCurrentTab,
    setPublicationStatusesForCurrentTab,
  ] = useState<PublicationStatus[]>(allowedPublicationStatuses);

const adminPublications = useAdminPublications({
    pageSize: 6,
    publicationStatusesForCurrentTab,
    rootLevelOnly: true,
    removeNotAllowedParents: true,
});
```

Say that `allowedPublicationStatuses` is `['published', 'archived']`. What is going to happen is

1. `useAdminPublicationsStatusCounts` is going to fetch the status counts for these allowed statuses
2. `useAdminPublications` is going to fetch all admin publications for these allowed statuses
3. When the response to the request done by `useAdminPublicationsStatusCounts` is in a bit later, it will see that actually only needs the published ones because that is what will be shown in the selected 'Active' tab
4. `useAdminPublications` will make another request, fetching now only the published admin publications.

This is the problem I avoid by putting them in different components

Reply:
Then we can use `[currentTab]` as the initial value of publicationStatusesForCurrentTab instead of `allowedPublicationStatuses`. That would solve your issue, right?

Reply:
No, because `currentTab` is not available yet- it can only be determined once we get the response from `useAdminPublicationsStatusCounts`